modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -129,6 +129,146 @@ G_TYPED_KERNEL(GBackgroundSubtractor, <GMat(GMat, BackgroundSubtractorParams)>,
     }
 };
 
+/** @brief Structure for the Kalman filter's initialization parameters.*/
+
+struct GAPI_EXPORTS KalmanParams
+{
+    //! Type of the created matrices that should be CV_32F or CV_64F.
+    int type = CV_32FC1;
+
+    // initial state
+
+    //! predicted state(x'(k)): x(k)=A*x(k-1)+B*u(k)
+    Mat statePre;
+    //! priori error estimate covariance matrix (P'(k)): P'(k)=A*P(k-1)*At + Q)*/
+    Mat errorCovPre;
+    //! corrected state (x(k)): x(k)=x'(k)+K(k)*(z(k)-H*x'(k))
+    Mat statePost;
+    //! posteriori error estimate covariance matrix (P(k)): P(k)=(I-K(k)*H)*P'(k)
+    Mat errorCovPost;
+
+    // dynamic system description
+
+    //! state transition matrix (A)
+    Mat transitionMatrix;
+    //! measurement matrix (H)
+    Mat measurementMatrix;
+    //! process noise covariance matrix (Q)
+    Mat processNoiseCov;
+    //! measurement noise covariance matrix (R)
+    Mat measurementNoiseCov;
+    //! control matrix (B) (Optional: not used if there's no control)
+    Mat controlMatrix;
+};
+
+G_TYPED_KERNEL(GKalmanFilter, <GMat(GMat, GOpaque<bool>, GMat, KalmanParams)>,
+               "org.opencv.video.KalmanFilter")
+{
+    static GMatDesc outMeta(const GMatDesc& measurement, const GOpaqueDesc&,
+                            const GMatDesc& control, const KalmanParams& kfParams)
+    {
+        GAPI_Assert(kfParams.type == CV_32FC1 || kfParams.type == CV_64FC1);
+        GAPI_Assert(!kfParams.controlMatrix.empty() && "controlMatrix should be initialized in this overload!");
```

Comment:
I'd cache dims and type here to simplify the checks below:

```
auto type = kfParams.transitionMatrix.type;
auto ddim = kfParams.transitionMatrix.cols;
auto mdim = kfParams.measurementMatrix.cols;
// ...
```

Reply:
Done.